logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamImpl.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.impl.log;
+
+import io.zeebe.dispatcher.Dispatcher;
+import io.zeebe.dispatcher.Dispatchers;
+import io.zeebe.dispatcher.impl.PositionUtil;
+import io.zeebe.logstreams.impl.Loggers;
+import io.zeebe.logstreams.log.LogStream;
+import io.zeebe.logstreams.log.LogStreamBatchWriter;
+import io.zeebe.logstreams.log.LogStreamReader;
+import io.zeebe.logstreams.log.LogStreamRecordWriter;
+import io.zeebe.logstreams.spi.LogStorage;
+import io.zeebe.util.ByteValue;
+import io.zeebe.util.sched.Actor;
+import io.zeebe.util.sched.ActorCondition;
+import io.zeebe.util.sched.ActorScheduler;
+import io.zeebe.util.sched.channel.ActorConditions;
+import io.zeebe.util.sched.future.ActorFuture;
+import io.zeebe.util.sched.future.CompletableActorFuture;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.agrona.concurrent.status.Position;
+import org.slf4j.Logger;
+
+public class LogStreamImpl extends Actor implements LogStream, AutoCloseable {
+  public static final long INVALID_ADDRESS = -1L;
+
+  private static final Logger LOG = Loggers.LOGSTREAMS_LOGGER;
+  private static final String APPENDER_SUBSCRIPTION_NAME = "appender";
+
+  private final ActorConditions onCommitPositionUpdatedConditions;
+  private final String logName;
+  private final int partitionId;
+  private final ByteValue maxFrameLength;
+  private final Position commitPosition;
```

Comment:
Do we still need a class with volatile semantic? Can't we just use a long now that the `LogStream` is an actor?

Reply:
we will use long now